src/Components/App/App.tsx

```diff
@@ -45,11 +49,27 @@ class App extends Component<{}, IAppState> {
       })
   }
 
+  findPostsWithSearchTerm = (searchTerm: string) => {
+    console.log("SEARCH TERM: ",searchTerm)
+
+    // this is only finding the first one, not sure why
+    return this.setState({ foundPosts: this.state.allPosts.filter(post => post.content.includes(searchTerm))} )
+
+  }
+
   renderComponent = () => {
-    if (this.state.allPosts.length > 0) {
+    if (this.state.foundPosts.length > 0) {
+      return (
+          <section className='main-page'>
+            <Search findPostsWithSearchTerm={this.findPostsWithSearchTerm} />
+            <AllPosts allPosts={this.state.foundPosts} />
+            <Nav />
+          </section>
+      )
```

Comment:
This sets the foundPosts as the preferred array and if that's empty, then it rolls down to allPosts.